src/Context/Modal/ModalProvider.jsx

```diff
@@ -0,0 +1,25 @@
+import { createContext, useContext, useState } from "react";
+
+const modalContext = createContext(null);
+
+const ModalProvider = ({ children }) => {
+  const [showLogin, setShowLogin] = useState(false);
+  const [showSignup, setShowSignup] = useState(false);
```

Comment:
Suggestion: We could provide more specific name here, instead of ModalProvider, something like "AuthModalProvider" as it is specifically being used for wrrapping authentication.